ui/src/app/views/new-notebook-modal/component.ts

```diff
@@ -44,10 +46,16 @@ export class NewNotebookModalComponent implements OnDestroy {
   }
 
   submitNewNotebook(): void {
-    const nbUrl = `/workspaces/` + encodeURIComponent(this.workspace.namespace) + '/' +
-        encodeURIComponent(this.workspace.id) + '/notebooks/create/?notebook-name=' +
-        encodeURIComponent(this.newName) + '&kernel-type=' +
-        encodeURIComponent(this.kernelType.toString());
+    const existingNotebook =
+      this.existingNotebooks.find((currentNotebook) =>
+        currentNotebook.name === this.newName + '.ipynb');
+    if (existingNotebook !== undefined) {
```

Comment:
nit: I think `!existingNotebook` is more readable

Reply:
(I think you mean `existingNotebook`, but agreed.